utopia-remix/app/util/api.server.spec.ts

```diff
@@ -52,3 +52,48 @@ describe('requireUser', () => {
     expect(got.name).toBe('Bob Bobson')
   })
 })
+
+describe('handle', () => {
+  it('runs handler', async () => {
+    const request = newTestRequest({ method: 'POST' })
+    const response = await (handle(
+      { request: request, params: {} },
+      {
+        POST: {
+          handler: async () => {
+            return { data: 'success' }
+          },
+          validator: async () => {
+            // passes
+          },
+        },
+      },
+    ) as Promise<ApiResponse<{ data: string }>>)
+    const result = await response.json()
+    expect(result).toEqual({
+      data: 'success',
+    })
+  })
+  it('fails on validator', async () => {
+    const request = newTestRequest({ method: 'POST' })
+    const response = await (handle(
+      { request: request, params: {} },
+      {
+        POST: {
+          handler: async () => {
+            return { data: 'success' }
+          },
+          validator: async () => {
+            throw new ApiError('failed', 400)
```

Comment:
for consistency, could you use `Status.[THE METHOD]` instead of hardcoded numbers when dealing with http status codes?

Reply:
fixed